components/map/Map.tsx

```diff
@@ -151,14 +152,19 @@ const Map = forwardRef(
             animated
           />
           {udrData && <MapZones udrData={udrData} />}
-          {selectedPolygon && (
-            <ShapeSource id="highlight" shape={selectedPolygon}>
-              <FillLayer
-                id="highlight"
-                style={udrStyle.find((layerStyle) => layerStyle.id === 'udr-fill-selected')?.paint}
-              />
-            </ShapeSource>
-          )}
+          <ShapeSource
+            id="highlight"
+            shape={selectedPolygon ?? { coordinates: [], type: 'Polygon' }}
```

Comment:
Some explanation in comments would make it easier to understand what is happening here :)